File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
const isPrime = (n) => {
    if(n < 2) return false;
    for(let i = 2; i <= Math.sqrt(n); i++) {
        if(n % i === 0) return false;
    }
    return true;
}
function solution(n, k) {
    const kNum = n.toString(k);
    const arr = kNum.split('0');
    if(arr.length === 1 && isPrime(kNum)) return 1;

    let answer = 0;
    for(let i = 0; i < arr.length; i++) {
        
        if(isPrime(Number(arr[i]))) answer++;
    }
    return answer;
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math

def solution(n, k):
    def convert_to_base(n, k):
        if n == 0:
            return "0"
        digits = []
        while n > 0:
            digits.append(str(n % k))
            n //= k
        return ''.join(reversed(digits))
    
    def is_prime(n):
        if n <= 1:
            return False
        if n == 2:
            return True
        for i in range(3, int(math.sqrt(n)) + 1):
            if n % i == 0:
                return False
        return True
    
    converted_n = convert_to_base(n, k)
    candidates = [int(x) for x in converted_n.split('0') if x != '' and is_prime(int(x))]
    
    return len(candidates)
